src/ostorlab/apis/assets.py

```diff
@@ -0,0 +1,189 @@
+        """
+        return """
+            mutation CreateAsset($asset: AssetInputType!){
+                createAsset(asset: $asset) {
+                    asset{
+                        ... on AndroidAssetType{
+                            id
+                        }
+                        ... on IOSAssetType{
+                            id
+                        }
+                        ... on AndroidFileAssetType{
+                            id
+                        }
+                        ... on IOSFileAssetType{
+                            id
+                        }
+                        ... on AndroidStoreAssetType{
+                            id
+                        }
+                        ... on IOSStoreAssetType{
+                            id
+                        }
+                        ... on UrlAssetType{
+                            id
+                        }
+                        ... on DomainAssetType{
+                            id
+                        }
+                        ... on SourceCodeAssetType{
+                            id
+                        }
+                        ... on FileAssetType{
+                            id
+                        }
+                        ... on IPAssetType{
+                            id
+                        }
+                        ... on IPv4AssetType{
+                            id
+                        }
+                        ... on IPv6AssetType{
+                            id
+                        }
+                    }
+                }
+            }
+        """
+
+    @property
+    def data(self) -> Optional[Dict]:
+        """Sets the body of the API request.
+
+        Returns:
+            The body of the create asset request.
+        """
+        asset_type = type(self._asset).__name__
+
+        variables = {
+            'asset': {
+                'tags': []
+            }
+        }
+        asset_type_variables = self.__get_asset_variables(asset_type)
+        variables['asset'].update(asset_type_variables)
+
+        if asset_type in ('AndroidAab', 'AndroidApk', 'File', 'IOSIpa'):
+            map_variables = _get_map_variables(asset_type)
+            data = {
+                'operations': json.dumps({'query': self.query,
+                                          'variables': json.dumps(variables)}),
+                'map': json.dumps(map_variables)
+            }
+        else:
+            data = {
+                'query': self.query,
+                'variables': json.dumps(variables)
+            }
+        return data
+
+    @property
+    def files(self) -> Optional[Dict] :
+        """Sets the file for multipart upload of the API request.
+
+        Returns:
+            The file mapping of the create asset request.
+        """
+        asset_type = type(self._asset).__name__
+        if asset_type in ('AndroidAab', 'AndroidApk', 'File', 'IOSIpa'):
```

Comment:
```suggestion
        if any([ isinstance(self._asset, t) for t of [AndroidAab, AndroidApk, File, IOSIpa]):
```
Checking asset instance by type name is bad, super bad, horribly bad.